chat_server/blueprints/preferences.py

```diff
@@ -26,31 +26,49 @@
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-from fastapi import APIRouter, Request, Form
-from chat_server.server_utils.auth import get_current_user, login_required
+from fastapi import APIRouter, Form, Depends
+
+from chat_server.server_utils.api_dependencies import CurrentUserData
+from chat_server.server_utils.api_dependencies.validators.users import (
+    get_authorized_user,
+)
 from utils.database_utils.mongo_utils.queries.wrapper import MongoDocumentsAPI
 from utils.http_utils import respond
-from utils.logging_utils import LOG
 
 router = APIRouter(
     prefix="/preferences",
     responses={"404": {"description": "Unknown user"}},
 )
 
 
+@router.post("/update")
+async def update_settings(
+    minify_messages: str = Form("0"),
```

Comment:
> Consider handling `preferences_mapping` here so added preferences don't need updates to this method

I am a bit worried that if someone would (accidentally/intentionally) abuse this endpoint with some redundant (e.g. huge) data, it may cause us significant overhead, so I would keep it as it

Reply:
Would a class extending `pydantic.BaseModel` be sufficient for validating/sanitizing the input? I think in the future this will accept dozens of parameters so it would be easier if the data structure was defined centrally for the UI and API to agree.

Reply:
> Would a class extending `pydantic.BaseModel` be sufficient for validating/sanitizing the input? I think in the future this will accept dozens of parameters so it would be easier if the data structure was defined centrally for the UI and API to agree.

Good suggestion, refactored that in https://github.com/NeonGeckoCom/pyklatchat/pull/95/commits/5d50824067c389658ace7f0ca55b4b6277c1d3ad